### PR TITLE
refactor: remove getUserEndpoints on push

### PR DESCRIPTION
**Pull request by nikhname**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { pathManager, JSONUtilities } from 'amplify-cli-core';
+import _ from 'lodash';
+
+const projectBackendDirPath = pathManager.getBackendDirPath();
```
Comment (attilah) [inline packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts:6]: this should be inside the function.

File: packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  let pinpointTemplateFile: any = JSONUtilities.readJson(pinpointTemplateFilePath);
+  pinpointTemplateFile.Resources.CognitoUnauthPolicy.Properties.PolicyDocument.Statement[0].Action = [
+    'mobiletargeting:PutEvents',
+    'mobiletargeting:UpdateEndpoint',
+  ];
+  pinpointTemplateFile.Resources.CognitoAuthPolicy.Properties.PolicyDocument.Statement[0].Action = [
+    'mobiletargeting:PutEvents',
+    'mobiletargeting:UpdateEndpoint',
+  ];
+  const pinpointTemplateFileJSON = JSONUtilities.stringify(pinpointTemplateFile);
```
Comment (attilah) [inline packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts:21]: this sould be `JSONUtilities.writeJson`

File: packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import _ from 'lodash';
+
+const projectBackendDirPath = pathManager.getBackendDirPath();
+const pinpointTemplateFileName = 'pinpoint-cloudformation-template.json';
+
+export function removeGetUserEndpoints(resourceName) {
+  const resourceDirPath = path.join(projectBackendDirPath, 'analytics', resourceName);
+  const pinpointTemplateFilePath = path.join(resourceDirPath, pinpointTemplateFileName);
+  let pinpointTemplateFile: any = JSONUtilities.readJson(pinpointTemplateFilePath);
+  pinpointTemplateFile.Resources.CognitoUnauthPolicy.Properties.PolicyDocument.Statement[0].Action = [
```
Comment (attilah) [inline packages/amplify-cli/src/extensions/amplify-helpers/remove-pinpoint-policy.ts:13]: Loadsh should be used for safe get and set and also remove the entry we don't want not setting all the others.

File: packages/amplify-cli/src/extensions/amplify-helpers/resource-status.ts [hunk 4]
```diff
@@ -179,6 +180,9 @@ async function getResourcesToBeUpdated(amplifyMeta, currentAmplifyMeta, category
   await asyncForEach(Object.keys(amplifyMeta), async categoryName => {
     const categoryItem = amplifyMeta[categoryName];
     await asyncForEach(Object.keys(categoryItem), async resource => {
+      if (categoryName === 'analytics') {
```
Comment (attilah) [inline packages/amplify-cli/src/extensions/amplify-helpers/resource-status.ts:183]: This is not right. Removal should happen on push as we discussed? an `amplify status` should not change resources.

**Review by attilah — CHANGES_REQUESTED** [review]

File: packages/amplify-cli/src/extensions/amplify-helpers/resource-status.ts (hunk 4 above)
Comment (nikhname) [reply packages/amplify-cli/src/extensions/amplify-helpers/resource-status.ts:183]: Discussed offline

**Review by attilah — APPROVED** [review]

**Pull request merged**